app/src/main/java/com/tiptopgoodstudio/androidresources/ui/LaunchActivity.java

```diff
@@ -44,38 +48,67 @@ protected void onCreate(Bundle savedInstanceState) {
         mLoadingMessage = (TextView) findViewById(R.id.tv_loading);
         mProgressBar = (ProgressBar) findViewById(R.id.pb_loading_indicator);
 
-        try {
-            //TODO - Create a method in ViewModel to access the Repository if this is not recommended
-            // First load data from Firebase into local DB if required
-            ResourceRepository resourceRepository = new ResourceRepository(getApplication());
-            DatabaseReference dbReference = resourceRepository.loadDataFromFirebase();
+        checkConnectivityAndLoadResources();
 
-            dbReference.addListenerForSingleValueEvent(new ValueEventListener() {
+    }
+
+    private void checkConnectivityAndLoadResources() {
+        isOnline = isOnline();
+
+        if(isOnline == true) {
+            try {
+                //TODO - Create a method in ViewModel to access the Repository if this is not recommended
+                // First load data from Firebase into local DB if required
+                ResourceRepository resourceRepository = new ResourceRepository(getApplication());
+                DatabaseReference dbReference = resourceRepository.loadDataFromFirebase();
 
-                //When we are done with updating data from Firebase, go to MainActivity
-                @Override
-                public void onDataChange(DataSnapshot dataSnapshot) {
-                    startMainActivity();
-                }
+                dbReference.addListenerForSingleValueEvent(new ValueEventListener() {
 
-                @Override
-                public void onCancelled(DatabaseError databaseError) {
+                    //When we are done with updating data from Firebase, go to MainActivity
+                    @Override
+                    public void onDataChange(DataSnapshot dataSnapshot) {
+                        startMainActivity();
+                    }
 
-                }
-            });
+                    @Override
+                    public void onCancelled(DatabaseError databaseError) {
```

Comment:
why aren't we printing the error here?  Should we remove this method entirely if we're not printing the error?

Reply:
onCancelled means if the Firebase import is cancelled. Isn't it different from the case where the import can't even begin because of no internet connection?